odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,154 @@ def _validate_module_views(self, module):
                 view._check_xml()
             except Exception as e:
                 self.raise_view_error("Can't validate view:\n%s" % e, view.id)
+
+    def _fix_faulty_views(self, delete=False):
+        """ Use the _get_faulty_views and take appropriate actions to fix the issue """
+        for view in self:
+            view_id = view.id # allow to use even if view is deleted
+            faulty_views = view._get_faulty_views(view,self.env['ir.ui.view'])
+            if faulty_views:
+                self._cr.execute('SAVEPOINT end_validation')
+                faulty_views._clean_properly(delete)
+                self._cr.execute("select count(id) from ir_ui_view where id = %s", (view_id,))
+                cnt = self.env.cr.fetchone()[0]
+                if cnt == 0:
+                    # handle the case where view itself has been deleted
+                    _logger.info("View %s can be fixed by disabling/deleting following views: %s" % (view_id,faulty_views))
```

Comment:
How do you delete an already deleted view?

Reply:
Hi, thank you for your question,
However, I'm not sure I fully understand it.
The view may be deleted during the _clean_properly(delete).
It is why I first check if the view exist before checking it is valid.
Does it answer the question ? If not, can you please explicit a the question,
Thank you for the feedback

Reply:
@nvn-odoo I guess it just looks really strange having an if / elif branch that does exactly the same thing. I've never seen it before, especially when the else is supposedly a non reachable case.

Reply:
Hi, Thank you for the feedback, the code has been adapted,
Best regards